tests/prepdata/test_image_transform.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+import pytest
+from numpy import testing
+from scipy.io import loadmat
+
+from kale.prepdata.image_transform import mask_img_stack, reg_img_stack, rescale_img_stack
+
+gait = loadmat("tests/test_data/gait_gallery_data.mat")
+images = gait["fea3D"][..., :10]
+SCALES = [4, 8]
+
+
+def test_reg():
+    n_samples = images.shape[-1]
+    coords = np.ones((n_samples, 4))
+    coords[:, 2:] += 20
+    coords[1:, :] += np.random.random(size=(n_samples - 1, 4))
+    with pytest.raises(Exception):
+        reg_img_stack(images, coords[1:, :])
+    images_reg, max_dist = reg_img_stack(images, coords)
+    testing.assert_allclose(images_reg, images)
```

Comment:
I did not get fully your logic here and can only guess. How did you make the registered images to be close to the original? I though you randomly perturbed the coordinates of the n-1 (9) images. After registration, they are close. Is it because the random noise is of small value compared to the coords? You can explain to me with a voice message in WeChat or Skype. Thanks.

Reply:
Sorry for the poor readability. Yes, the random noise to source coordinates are in (0, 1), which are small values compared to the image size. So the images after registration should be close to the original ones. The values in list `max_dist` are very small. Maybe we can test whether `max_dist + 1` is close to a ones vector (`+1` for avoiding inf relative difference). Because we do not have real images and landmark coordinates for testing here, this is probably the easiest way for me to test the function. I will add more comments to improve the readability.

Reply:
Thanks for the clarification. That's helpful! Ready to merge.